cloud/spawn.go

```diff
@@ -69,14 +77,8 @@ func (so *SpawnOptions) validate(settings *evergreen.Settings) error {
 	}
 
 	// validate public key
-	rsa := "ssh-rsa"
-	dss := "ssh-dss"
-	isRSA := strings.HasPrefix(so.PublicKey, rsa)
-	isDSS := strings.HasPrefix(so.PublicKey, dss)
-	if !isRSA && !isDSS {
-		return errors.Errorf("Invalid spawn options: "+
-			"either an invalid Evergreen-managed key name has been provided,"+
-			"or the key value does not start with %s or %s", rsa, dss)
+	if err = ValidateSSHKey(so.PublicKey); err != nil {
```

Comment:
If we want this to be available to the CLI and to the server-side code, maybe move this into the top-level `evergreen` package? Otherwise, we would ahve to call this cloud provider validation code from the `operations` package.

Reply:
done